pkg/config/environment_detection.go

```diff
@@ -116,13 +118,39 @@ func detectContainerFeatures() {
 	}
 }
 
+func checkSocketExists(path string) bool {
+	f, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	// On Windows, we cannot easily verify that a path is a named pipe
+	if runtime.GOOS == "windows" {
+		return true
+	}
+
+	if !f.IsDir() && f.Mode()&os.ModeSocket > 0 {
+		return true
+	}
+
+	return false
+}
+
 func getHostMountPrefixes() []string {
 	if IsContainerized() {
 		return []string{"", defaultHostMountPrefix}
 	}
 	return []string{""}
 }
 
+func getDefaultDockerSocketType() string {
+	if runtime.GOOS == "windows" {
+		return "npipe://"
+	}
+
+	return "unix://"
```

Comment:
nit: These static strings seem to match the types of constants near the top of the file  so they can easily be pulled up there